hbase-client/src/main/java/org/apache/hadoop/hbase/util/ColumnFamilyAttributeValidator.java

```diff
@@ -0,0 +1,102 @@
+import org.apache.hadoop.hbase.MemoryCompactionPolicy;
+import org.apache.hadoop.hbase.client.MobCompactPartitionPolicy;
+import org.apache.hadoop.hbase.io.compress.Compression;
+import org.apache.hadoop.hbase.io.encoding.DataBlockEncoding;
+import org.apache.hadoop.hbase.regionserver.BloomType;
+import org.apache.yetus.audience.InterfaceAudience;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@InterfaceAudience.Private
+public class ColumnFamilyAttributeValidator {
+  private static final Logger LOG = LoggerFactory.getLogger(ColumnFamilyAttributeValidator.class);
+  private final static Map<Bytes, Function<Bytes,Object>> ATTRIBUTE_VALIDATOR = new HashMap<>();
+
+  static {
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(DATA_BLOCK_ENCODING)),
+      (d) -> DataBlockEncoding.valueOf(d.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(COMPRESSION)),
+      (ca) -> Compression.Algorithm.valueOf(ca.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(BLOOMFILTER)),
+      (bl) -> BloomType.valueOf(bl.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(REPLICATION_SCOPE)),
+      (rs) -> Integer.valueOf(rs.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(MIN_VERSIONS)),
+      (min) -> Integer.valueOf(min.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(MAX_VERSIONS)),
+      (max) -> Integer.valueOf(max.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(BLOCKSIZE)),
+      (bs) -> Integer.valueOf(bs.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(KEEP_DELETED_CELLS)),
+      (k) -> KeepDeletedCells.valueOf(k.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(TTL)),
+      (ttl) -> Integer.valueOf(ttl.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(DFS_REPLICATION)),
+      (dr) -> Integer.valueOf(dr.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(MOB_THRESHOLD)),
+      (mt) -> Long.valueOf(mt.toString()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(MOB_COMPACT_PARTITION_POLICY)),
+      (mc) -> MobCompactPartitionPolicy.valueOf(mc.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(IN_MEMORY_COMPACTION)),
+      (im) -> MemoryCompactionPolicy.valueOf(im.toString().toUpperCase()));
+    ATTRIBUTE_VALIDATOR.put(new Bytes(Bytes.toBytes(COMPRESSION_COMPACT)),
+      (ca) -> Compression.Algorithm.valueOf(ca.toString().toUpperCase()));
+  }
+
+  /**
+   * Validates values of HBase defined column family attributes.
+   * Throws {@link IllegalArgumentException} if the value of attribute is not proper format.
+   * @param key
+   * @param value
+   */
+  public static void validateAttributeValue(Bytes key, Bytes value) {
+    if(ATTRIBUTE_VALIDATOR.containsKey(key)) {
+      try {
+        ATTRIBUTE_VALIDATOR.get(key).apply(value);
```

Comment:
```
Function<Byte,Object> validator = ATTRIBUTE_VALIDATE.get(key);
if (validator != null) {
  try {
    validator.apply(value);
```

Would be a little nicer, IMO (only one map lookup).

Reply:
Thanks @joshelser  for review. Done